src/Controller/AppController.php

```diff
@@ -277,4 +278,51 @@ protected function applySessionFilter() : ?Response
 
         return null;
     }
+
+    /**
+     * Set objectNav array
+     *
+     * @param array $objects The objects to parse to set prev and next data
+     * @return void
+     */
+    protected function setObjectNav($objects) : void
+    {
+        $moduleName = $this->Modules->getConfig('currentModuleName');
+        $total = count(array_keys($objects));
+        $objectNav = [];
+        foreach ($objects as $i => $object) {
+            $objectNav[$moduleName][$object['id']] = [
+                'prev' => ($i > 0) ? Hash::get($objects, sprintf('%d.id', $i - 1)) : null,
+                'next' => ($i + 1 < $total) ? Hash::get($objects, sprintf('%d.id', $i + 1)) : null,
+                'index' => $i + 1,
+                'total' => $total,
+                'object_type' => Hash::get($objects, sprintf('%d.object_type', $i)),
+            ];
+        }
+        $session = $this->request->getSession();
+        $session->write('objectNav', $objectNav);
```

Comment:
Using `objectNav.$moduleName` as key? We don't need `objectTypeNav` and we could store more lists at the same time (e.g. navigate items in different tabs..)

Reply:
We need the info stored in `objectTypeNav`... it tells the view which was the module index recently used.
The `currentModuleName` is not a good choice... think about a navigation in "object" index, and then to a view of a "document".  In this case `objectTypeNav = 'objects'` and `currentModuleName = 'documents'`. The object navigation should use `'objects'`